syntax = "proto3";

package bluetooth.hci;

import "google/protobuf/empty.proto";
import "facade/common.proto";

service LeInitiatorAddressFacade {
  rpc SetPrivacyPolicyForInitiatorAddress(PrivacyPolicy) returns (google.protobuf.Empty) {}
  rpc GetCurrentInitiatorAddress(google.protobuf.Empty) returns (bluetooth.facade.BluetoothAddressWithType) {}
  rpc GetAnotherAddress(google.protobuf.Empty) returns (bluetooth.facade.BluetoothAddressWithType) {}
}

enum AddressPolicy {
  POLICY_NOT_SET = 0x00;
  USE_PUBLIC_ADDRESS = 0x01;
  USE_STATIC_ADDRESS = 0x02;
  USE_NON_RESOLVABLE_ADDRESS = 0x03;
  USE_RESOLVABLE_ADDRESS = 0x04;
}

message PrivacyPolicy {
  AddressPolicy address_policy = 1;
  facade.BluetoothAddressWithType address_with_type = 2;
  bytes rotation_irk = 3;
  uint64 minimum_rotation_time = 4;
  uint64 maximum_rotation_time = 5;
}
